flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorTest.java

```diff
@@ -101,9 +110,44 @@
                                     LocalDateTime.of(2020, 3, 19, 12, 22, 14, 1000)),
                             (int) LocalDate.of(2020, 3, 19).toEpochDay(),
                             (int) (LocalTime.of(12, 13, 14, 2000).toNanoOfDay() / 1_000_000L),
+                            TimestampData.fromInstant(
+                                    LocalDateTime.of(2020, 3, 19, 12, 22, 14, 1000)
+                                            .atZone(ZoneId.of("Asia/Shanghai"))
```

Comment:
nit: I would use another timezone here, maybe `America/Los_Angeles` which has a `-` offset from UTC.
Maybe also use an our > 12 closer to 24, like `20` to have a resulting date that "jumps ahead" (20/3 instead of 19/3).